spec/game_runner_spec.rb

```diff
@@ -0,0 +1,30 @@
+require_relative 'spec_helper'
+require_relative '../lib/game_runner'
+
+RSpec.describe GameRunner, "#run" do
+  def grid_format
+    (<<~GRID)
+   __ __ __
+  |  |  |  |
+  |__|__|__|
+  |  |  |  |
+  |__|__|__|
+  |  |  |  |
+  |__|__|__|
+    GRID
+  end
+
+  context "on start up" do
+    it "renders an empty grid on the screen" do
+      output = StringIO.new("it doesn't matter what goes here")
+      $stdout = output
+      mock_output = $stdout.string
+      game = GameRunner.new
+
+      game.run
+
+      expect(mock_output).to eq(grid_format)
```

Comment:
It seems like this is failing properly, now. What do you think?

Reply:
Yes and no. You never want to use global variables in Ruby. And you definitely don't want to override a global variable as important as standard out. That's why the instructions mention [dependency injection](https://www.sandimetz.com/blog/2009/03/21/solid-design-principles).

That has to change. Also I am still seeing code in implementation that hasn't been generated by a test need.

So, really, really, go back to zero and do a new branch, create a new PR with all the things you have learned. So you should only have code that you need to get a properly failing test. You shouldn't be overwriting global variables, but instead need to dependency inject the mock input and output.